app/models/product_import/entry_validator.rb

```diff
@@ -124,16 +124,15 @@ def producer_validation(entry)
     end
 
     def inventory_validation(entry)
-      # Checks a potential inventory item corresponds to a valid variant
-      match = Spree::Product.where(supplier_id: entry.producer_id, name: entry.name, deleted_at: nil).first
+      products = Spree::Product.where(supplier_id: entry.producer_id, name: entry.name, deleted_at: nil)
 
-      if match.nil?
+      if products.empty?
         mark_as_invalid(entry, attribute: 'name', error: I18n.t('admin.product_import.model.no_product'))
         return
       end
 
-      match.variants.each do |existing_variant|
-        unit_scale = match.variant_unit_scale
+      products.map(&:variants).flatten.each do |existing_variant|
```

Comment:
I feel this can be simpler with [#flat_map](https://apidock.com/ruby/Enumerable/flat_map)

Reply:
Thanks, I didn't know about #flat_map